guides/terminology.md

```diff
@@ -0,0 +1,89 @@
+## Terminology
+
+### Mutation Backflow
+
+Mutation backflow occurs when a child context mutates an object that is owned
+by a parent context, causing changes in the parent context.  This is analogous
+to "spooky action at a distance".
+ 
+An example negative side-effect of mutation backflow occurs when a parent
+component has already rendered when a child component mutates an object owned
+or utilized by the parent component.  This forces the parent component to need
+to re-render, and is one of several causes of what we refer to as a backtracking
+rerender.
+
+### Backtracking Rerender
+
+A backtracking rerender occurs when already rendered data is mutated before the
+entire render completes. This forces a second render pass to account for any
+updated state. Backtracking rerenders are synchronous and will block paint.
+
+### The State Tree
+
+In a web application, state can live at many different scopes. We can represent
+the scopes from outermost to innermost as a tree.
+ 
+* Global State (state that is universal, would be true for any number of
+  applications on the same window, such as in SSR or when running tests)
+  * Application State (state that should live and die with the application
+    instance, the application route and services fall into this category)
+    * Route State (state that should live and die based on a route 
+      activating / deactivating)
+      * Local State (state that should live and die based on the display
+        lifecycle of a single component, also known as "UI State" or
+        "Component State")
```

Comment:
While this tree makes sense in a variety of situations, I'm not comfortable with just tossing this out there with a couple of sentences of explanation. State management is one of the biggest challenges that developers have with their apps, and incomplete/cursory guidance in this area can be worse than no guidance at all.